addons/website_project/tests/test_website_project.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.tests.common import HttpCase, tagged
+
+
+@tagged('-at_install', 'post_install')
+class TestWebsiteProject(HttpCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.test_project = cls.env['project.project'].create({'name': 'Project_1'})
+        cls.test_partner = cls.env['res.partner'].create({
+            'name': 'Test Partner',
+            'email': 'test@partner.com',
+        })
+
+    def test_portal_task_submission(self):
+        """ Public user should be able to submit a task"""
+        self.authenticate(None, None)
+        task_data = {
+            'name': "test_task_portal",
+            'email_from': 'test@test.com',
+            'description': 'This test task is created by Portal',
+            'project_id': self.test_project.id,
+            'csrf_token': http.Request.csrf_token(self),
+        }
+        response = self.url_open('/website/form/project.task', data=task_data)
+        task = self.env['project.task'].browse(response.json().get('id'))
+        self.assertTrue(task.exists())
+        self.assertFalse(task.partner_id, "Partner id should be False")
+        self.assertEqual(task.email_cc, 'test@test.com', "email_cc should be same as added on website")
+
+        mail_message = task.message_ids.filtered(lambda m: m.body == '<div class="alert alert-info">/!\\ EXTERNAL SUBMISSION - Customer not verified</div>')
+        self.assertIn(mail_message, task.message_ids, "Alert Message should be display in chatter")
+
+    def test_admin_task_submission(self):
+        """ Admin should be able to submit a task"""
+        self.authenticate("admin", "admin")
+        task_data = {
+            'name': "test_task_admin",
+            'email_from': 'test@partner.com',
+            'description': 'This test task is created by Admin',
+            'project_id': self.test_project.id,
+            'csrf_token': http.Request.csrf_token(self),
+        }
+        response = self.url_open('/website/form/project.task', data=task_data)
+        task = self.env['project.task'].browse(response.json().get('id'))
+        self.assertTrue(task.exists())
+        self.assertEqual(task.partner_id, self.test_partner, "Partner id should not be False")
+        self.assertFalse(task.email_cc, "email_cc field should be empty")
+
+        mail_message = task.message_ids.filtered(lambda m: m.body == '<div class="alert alert-info">This Task was submitted by Mitchell Admin (admin@yourcompany.example.com) on behalf of test@partner.com</div>')
+        self.assertIn(mail_message, task.message_ids, "Alert Message should be display in chatter")
```

Comment:
Since we did the filtered on `message_ids` this assert is a bit useless.

We should check we found the message and we only got 1 message.

```suggestion
        self.assertEqual(len(mail_message), 1, "Alert message should be displayed in the chatter of the task created.")
```

Reply:
Hello @xavierbol 
All necessary changes are done.

Thank You :slightly_smiling_face: